department-of-veterans-affairs/va.gov-cms
# VACMS-3693: Adds userSection to GTM dataLayer.

## Description

See #3693. 

I need to add a further test to check for and warn of conditions where any user has a number of discrete sections enabled such that they would overflow the length limit of the value sent to GA.  At some point this will be converted into a more appropriate form, e.g. a check on the workplace_access form, a warning on a CMS dashboard, or something similar.

This is also blocked by some work coordinating with other teams.

## Testing done
Added Behat tests.

## Screenshots
<img width="313" alt="Screen Shot 2021-01-13 at 11 21 03 AM" src="https://user-images.githubusercontent.com/1318579/104479460-74047480-5591-11eb-9347-7e3172672eca.png">

## QA steps
As any `administrator` user:
1. Browse to any CMS page and inspect the `dataLayer` variable in Developer Tools
1. Then validate Acceptance Criteria:
    - [ ] userSection should have a null value by default for administrators
    - [ ] checking various values on the Workbench Access tab of your user profile (and saving, obv.) should have effects on this value:
        - [ ] checking "Sections" should change this value to `all`
        - [ ] checking various different sections should change this value to a comma-separated list of tids
As any `content_admin` user:
1. Browse to any CMS page and inspect the `dataLayer` variable in Developer Tools
1. Then validate Acceptance Criteria:
    - [ ] userSection value should be `all`

### Definition of Done


### Select Team for PR review


### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
